Tasks/AzureFileCopy/AzureUtilityGTE1.0.ps1

```diff
@@ -340,4 +340,108 @@ function Remove-AzureMachineCustomScriptExtension
     }
 
     return $response
+}
+
+function Add-NetworkSecurityRuleConfig
+{
+    param([object]$securityGroup,
+          [string]$ruleName,
+          [string]$rulePriotity,
+          [string]$winrmHttpsPort)
+
+    if($securityGroup)
+    {
+        $securityGroupName = $securityGroup.Name
+        try
+        {
+            Write-Verbose -Verbose "[Azure Call]Getting network security rule config $ruleName under security group $securityGroupName"
+            $winRMConfigRule = Get-AzureRmNetworkSecurityRuleConfig -NetworkSecurityGroup $securityGroup -Name $ruleName -EA SilentlyContinue
+            Write-Verbose -Verbose "[Azure Call]Got network security rule config $ruleName under security group $securityGroupName"
+        }
+        catch
+        {
+            #Ignore the exception
```

Comment:
Should you return right here? Else you will proceed further to print message that rule already exists.